### DISCUSSION TITLE
understanding `/mode expert` complications on servers

### DISCUSSION THREAD

**Discussion by Quezler** [NillerMedDild/Enigmatica6#2752] [Q&A] [answered]
https://github.com/NillerMedDild/Enigmatica6/blob/852484c442d430d7a01b510fb4f201da3d3dfdcc/config/ftbquests/quests/chapters/getting_started.snbt#L166

this is currently the "warning" before switching mode, this seems self explanatory enough for singleplayer games, but what about multiplayer?

- does every player that freshly connects need to restart their game so the "config changes" can affect them?
- does it even matter at all for players connecting to a server, since it does mention config changes?
- if player joined an expert server (and then restarted & rejoined), and then joined a single/multi player map that is on normal mode (without restarting prior or after), would that cause issues?
- what conflicts/issues can arise currently from ignoring the reboot instructions? 

its quite possible that i'm just overthinking it, but a little clarification would be most welcome 🤔

**Comment by Darkere** [comment]
This really depends on the config specifically. Most configs will be serverconfigs or datapack/script stuff that will reload on world join anyways.

But there can be some configs that do not, and for those you'd need a restart. The type of problems that can occur also depend on the specific config so ... no idea :/ .. Anything from crash to less serious stuff.

**Comment by NielsPilgaard** [comment, marked as answer]
You aren't able to see that quest by default right? :P

> does every player that freshly connects need to restart their game so the "config changes" can affect them?

Yes, preferably.

> does it even matter at all for players connecting to a server, since it does mention config changes?

Yes, the player has to be in the same mode when joining the server.

> if player joined an expert server (and then restarted & rejoined), and then joined a single/multi player map that is on normal mode (without restarting prior or after), would that cause issues?

It could, yes. Both server and client need to be in the same mode.

**Reply by Quezler** [reply]
> need to be in the same mode.

is connecting to a server, restarting the game & rejoining sufficient enough to "set" the client to the mode of the server / last joined map? (aka: is the server (or any joined map for that matter) able to chance the "startup" config files of the client?)

**Reply by Quezler** [reply]
> You aren't able to see that quest by default right? :P

no idea, didn't check, someone on the discord mentioned /mode and it peeked my interest, started digging with github search 😅

**Reply by NielsPilgaard** [reply]
I don't think it is, no, they'd have to run the command in singleplayer, or change the file manually. I could totally be wrong though, @Darkere help 😄

**Reply by Quezler** [reply]
ah yes, i see it now (didn't look for it before), appears to be in the main minecraft folder (not any config/script subfolders):

<img width="516" alt="Screen Shot 2021-07-16 at 20 57 18" src="https://user-images.githubusercontent.com/3179271/125995933-ef375772-4af1-44c0-9753-d9f49c488d3e.png">

(i'm no modpack expert, but if this acts the way it is described a sort of motd warning when joining a server with a different mode prompting the user to modify it "somehow" sounds like a welcome feature, still no idea what exactly it could break 🤔)

**Reply by NielsPilgaard** [reply]
Yeah you're right ^^'
